File: modules/widgets/lib/rin/widgets/event/CursorUpEvent.cpp
#include "rin/widgets/event/CursorUpEvent.hpp"

CursorUpEvent::CursorUpEvent(const Shared<WidgetSurface>& inSurface, const CursorButton inButton,
        const Vec2<float>& inPosition) : Event(inSurface), button(inButton), position(inPosition)
{
        
}

File: rin.Widgets/Events/CursorUpEvent.cs
using rin.Core.Math;
using rin.Graphics.Windows;
using rin.Widgets.Graphics;
using rin.Windows;

namespace rin.Widgets.Events;

public class CursorUpEvent(Surface surface, CursorButton button, Vector2<float> position)
    : Event(surface)
{
    public Vector2<float> Position = position;
    public CursorButton Button = button;
}